internal/pkg/coordinator/monitor.go

```diff
@@ -107,10 +107,24 @@ func (m *monitorT) Run(ctx context.Context) (err error) {
 		return ctx.Err()
 	}
 
+	// Start timer loop to ensure leadership
+	lT := time.NewTimer(m.checkInterval)
+	defer lT.Stop()
+
 	// Ensure leadership on startup
-	err = m.ensureLeadership(ctx)
-	if err != nil {
-		return err
+	for {
```

Comment:
The coordinator and monitors get started  separate goroutines ( in `server/fleet.go`) so I don't think anything should go wrong with these changes.
The only difference should be that the API is "available", but should return a 503 in a non-status endpoint (like enroll) is called. Can you add that as an e2e test?

Reply:
And removed after https://github.com/elastic/fleet-server/pull/2693#discussion_r1240095377 and the discussion about health checks.

Not sure then if there is an E2E test that we can add here, maybe in the controllers that use it?